There is something wonky in the dependency genreation:
      khotkeys-5.10.95/app/kded.cpp:31: Error: Plugin Metadata
      file "khotkeys.json" does not exist. Declaration will be ignored

That file should be generated by a custom command, which the target in question
already depends on. As this is only a test application that is not even
installed, simply ignore it.

--- CMakeLists.txt.orig	2017-09-27 08:23:50 UTC
+++ CMakeLists.txt
@@ -55,7 +55,6 @@ add_subdirectory( app )
 add_subdirectory( kcm_hotkeys )
 # kdedmodule and standalone demon for hotkeys
 add_subdirectory( data )
-add_subdirectory( test )
 add_subdirectory( doc )
 feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)

